model/host/host.go

```diff
@@ -1600,6 +1600,37 @@ func (h *Host) AddSSHKeyName(name string) error {
 	return nil
 }
 
+// HandleTerminatedHostSpawnedByTask should be called when a host spawned by a task
+// is found terminated.
+func (h *Host) HandleTerminatedHostSpawnedByTask() error {
+	if !h.SpawnOptions.SpawnedByTask {
+		return nil
+	}
+
+	t, err := task.FindOneIdAndExecution(h.SpawnOptions.TaskID, h.SpawnOptions.TaskExecutionNumber)
+	if err != nil {
+		return errors.Wrapf(err, "finding task for host '%s'", h.Id)
+	}
+	if t == nil {
+		return errors.Errorf("host '%s' created by task '%s' execution '%d' that does not exist", h.Id, h.SpawnOptions.TaskID, h.SpawnOptions.TaskExecutionNumber)
+	}
+
+	if h.Status == evergreen.HostStarting && t.Status == evergreen.TaskStarted && !t.Aborted {
+		intent := newIntentFromHost(h)
+		grip.Info(message.Fields{
+			"message":              "inserting a host intent to replace a terminated host for a task",
+			"original_host":        h.Id,
+			"original_host_status": h.Status,
+			"new_host":             intent.Id,
+			"task":                 t.Id,
+			"task_execution":       t.Execution,
+		})
+		return intent.Insert()
```

Comment:
It seems like if this insertion fails, host.create will be considered a failure since it won't get another host. It seems like this should also call `AddHostCreateDetails` in the insertion error case.

Same with the other error cases above this - if they're not transient errors and this method does not retry (at the Amboy job level), it seems like they should populate the host create details.

Reply:
I think I've done this now. 
Though if the db call to get the task fails perhaps that's not transient and we could retry 🤷